libretroshare/src/gxstrans/p3gxstransitems.cc

```diff
@@ -48,7 +48,9 @@ void OutgoingRecord_deprecated::serial_process(RsGenericSerializer::SerializeJob
 	RS_REGISTER_SERIAL_MEMBER(recipient);
 	RS_REGISTER_SERIAL_MEMBER(mailItem);
 	RS_REGISTER_SERIAL_MEMBER(mailData);
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 	RS_REGISTER_SERIAL_MEMBER_TYPED(clientService, uint16_t);
```

Comment:
This warning tells something concerning. We should not hide it, but rather fix the wrong cast below.

Reply:
Done, only remove them, now no warnings???

Reply:
I would not concern much about those casts as PR #1155 introduce a new system that doesn't suffer of that kind of problems

Reply:
Do you says that all of these code could be removed?

Reply:
no absolutely!! I say that those warning disappears when https://github.com/RetroShare/RetroShare/pull/1155 will be merged as i have found a better way to handle enums in serialization there